reasoner/controller/DisjunctionController.java

```diff
@@ -45,16 +45,16 @@
 import static com.vaticle.typedb.core.common.iterator.Iterators.iterate;
 import static com.vaticle.typedb.core.reasoner.controller.ConjunctionController.merge;
 
-public abstract class DisjunctionController<
-        PROCESSOR extends DisjunctionController.Processor<PROCESSOR>,
-        CONTROLLER extends DisjunctionController<PROCESSOR, CONTROLLER>
-        > extends AbstractController<ConceptMap, ConceptMap, ConceptMap, Request, PROCESSOR, CONTROLLER> {
+public abstract class DisjunctionController<OUTPUT,
+        PROCESSOR extends DisjunctionController.Processor<OUTPUT, PROCESSOR>,
+        CONTROLLER extends DisjunctionController<OUTPUT, PROCESSOR, CONTROLLER>
+        > extends AbstractController<ConceptMap, ConceptMap, OUTPUT, Request, PROCESSOR, CONTROLLER> {
```

Comment:
Make DisjunctionController's output type generic, so we can let ConditionController subclass it.